project/project.go

```diff
@@ -572,6 +578,47 @@ func SplitTest(name string) (string, string) {
 	return name, ""
 }
 
+// DoesTestcaseMatchPreset: check whether testcase instance shall
+// be executed dependent on the specified presets
+func DoesTestcaseMatchPreset(tc *TestConfig, presets []string) bool {
```

Comment:
I would split this function into smaller ones with less complexity.
Something like bellow, if you know what I mean:

```go
func DoesTestConfigMatchPresets(tc TestConfig, presets ...string) bool {
    if tc.Except == nil && tc.Only == nil {
        return true
    }
    return !MatchPresets(tc.Exept, presets...) && MatchPresets(c.Only, presets...)
}

func MatchPresets(c *ExecuteCondition, preset ...string) bool {
    for _, p := range presets {
        for _, q := range c.Presets {
            if p == q {
                return true
            }
        }    
    }
   return false
}
```